src/modules/cpu/kernel/exclusive_or.hpp

```diff
@@ -0,0 +1,965 @@
+            dstPtrRow = dstPtrChannel;
+
+            for (int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtr1TempR, *srcPtr1TempG, *srcPtr1TempB, *srcPtr2TempR, *srcPtr2TempG, *srcPtr2TempB, *dstPtrTemp;
+                srcPtr1TempR = srcPtr1RowR;
+                srcPtr1TempG = srcPtr1RowG;
+                srcPtr1TempB = srcPtr1RowB;
+                srcPtr2TempR = srcPtr2RowR;
+                srcPtr2TempG = srcPtr2RowG;
+                srcPtr2TempB = srcPtr2RowB;
+                dstPtrTemp = dstPtrRow;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m256i p1[3], p2[3];
+
+                    rpp_simd_load(rpp_load96_u8pln3_to_u8pln3, srcPtr1TempR, srcPtr1TempG, srcPtr1TempB, p1);    // simd loads
+                    rpp_simd_load(rpp_load96_u8pln3_to_u8pln3, srcPtr2TempR, srcPtr2TempG, srcPtr2TempB, p2);    // simd loads
+                    p1[0] = _mm256_xor_si256(p1[0], p2[0]);    // exclusive_or computation
+                    p1[1] = _mm256_xor_si256(p1[1], p2[1]);    // exclusive_or computation
+                    p1[2] = _mm256_xor_si256(p1[2], p2[2]);    // exclusive_or computation
+                    rpp_simd_store(rpp_store96_u8pln3_to_u8pkd3, dstPtrTemp, p1);    // simd stores
+
+                    srcPtr1TempR += vectorIncrementPerChannel;
+                    srcPtr1TempG += vectorIncrementPerChannel;
+                    srcPtr1TempB += vectorIncrementPerChannel;
+                    srcPtr2TempR += vectorIncrementPerChannel;
+                    srcPtr2TempG += vectorIncrementPerChannel;
+                    srcPtr2TempB += vectorIncrementPerChannel;
+                    dstPtrTemp += vectorIncrement;
+                }
+
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    dstPtrTemp[0] = *srcPtr1TempR ^ *srcPtr2TempR;
+                    dstPtrTemp[1] = *srcPtr1TempG ^ *srcPtr2TempG;
+                    dstPtrTemp[2] = *srcPtr1TempB ^ *srcPtr2TempB;
+
+                    srcPtr1TempR++;
+                    srcPtr1TempG++;
+                    srcPtr1TempB++;
+                    srcPtr2TempR++;
+                    srcPtr2TempG++;
+                    srcPtr2TempB++;
+                    dstPtrTemp += 3;
+                }
+
+                srcPtr1RowR += srcDescPtr->strides.hStride;
+                srcPtr1RowG += srcDescPtr->strides.hStride;
+                srcPtr1RowB += srcDescPtr->strides.hStride;
+                srcPtr2RowR += srcDescPtr->strides.hStride;
+                srcPtr2RowG += srcDescPtr->strides.hStride;
+                srcPtr2RowB += srcDescPtr->strides.hStride;
+                dstPtrRow += dstDescPtr->strides.hStride;
+            }
+        }
+
+        // Exclusive OR without fused output-layout toggle (NHWC -> NHWC or NCHW -> NCHW)
+        else
```

Comment:
please try a different code for pkd3 -pkd3 conversion , please try to use 96 elements load and store routines.

Applicable for all the bitdepths

Reply:
96 elements load and store use
Separate function written for PLN3 to PLN3 based on discussion
PKD3 to PKD3 and PLN1 to PLN1 same code, channel loop removed